### PR TITLE
Feature/card input as hook

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
Move CardInput comp to Preact hooks 

## Tested scenarios
All e2e tests pass
All unit tests pass

**Fixed issue**:  COWEB-950

### CODE REVIEW

File: packages/lib/src/components/internal/SecuredFields/lib/configuration/constants.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const GIFT_CARD = 'giftcard';
 export const ENCRYPTED_BANK_ACCNT_NUMBER_FIELD = 'encryptedBankAccountNumber';
 export const ENCRYPTED_BANK_LOCATION_FIELD = 'encryptedBankLocationId';
 
-export const SF_VERSION = '3.5.1';
+export const SF_VERSION = '3.5.0';
```
Comment (marcperez) [inline packages/lib/src/components/internal/SecuredFields/lib/configuration/constants.ts:17]: Is this accidental?

Comment (sponglord) [reply packages/lib/src/components/internal/SecuredFields/lib/configuration/constants.ts:17]: No. 3.5.1 isn't on test yet

Comment (sponglord) [reply packages/lib/src/components/internal/SecuredFields/lib/configuration/constants.ts:17]: I take that back - it is now!

File: packages/lib/src/components/Card/components/CardInput/CardInput.tsx [hunk 2]
```diff
@@ -1,330 +1,398 @@
...
+        if (billingAddressRef?.current) billingAddressRef.current.showValidation();
+    };
+
+    this.processBinLookupResponse = (binLookupResponse: BinLookupResponse, isReset: boolean) => {
+        extensions.processBinLookup(binLookupResponse, isReset);
+    };
+
+    this.setStatus = status => {
+        setStatus(status);
+    };
```
Comment (marcperez) [inline packages/lib/src/components/Card/components/CardInput/CardInput.tsx:173]:
```suggestion
    this.setStatus = setStatus;
```

Very minor, could be simplified to this.

**Review by marcperez — APPROVED** [review]

**Pull request merged**